nemo/collections/tts/models/speechllm/megatron_t5_speechllm_model.py

```diff
@@ -1416,14 +1416,14 @@ def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> A
             end_inference_loop_at = None
             fwd_bwd_function = get_forward_backward_func()
             encoder_output = None
-            for t in range(self.decoder_context_len, dec_input.shape[2] - 1):
+            for t in range(self.decoder_context_len + 1, dec_input.shape[2] - 1):
                 # Start at 0 if encoder context, else context_len
                 if t % 100 == 0:
                     logging.info("Timestep {}".format(t))
                 if t == end_inference_loop_at:
                     print("All ends detected")
                     break
-                if t == self.decoder_context_len:
+                if t == self.decoder_context_len + 1:
```

Comment:
nit: sanity check, is `+1` a typo?

Reply:
I think this is because, we need an extra token after decoder context (which is a padding token on which loss is masked during training). I think there is a zero padding between context and answer, and during inference we need to pass that padding as input to correctly start predicting the answer.